### PR TITLE
[RDPP 4451] Fix inner join and partially switch on the test suite

### PR DESCRIPTION
**Pull request by pltod2**

### CODE REVIEW

File: test/batched.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ describe('Connector', function () {
 	var Models = common.Models,
 		IDs = common.IDs;
 
-	it('should be able to batched find all', function (next) {
```
Comment (todanator) [inline test/batched.js]: You can put a "skip" on the "describe" instead of all of these

File: test/common.js [hunk 2]
```diff
@@ -18,80 +18,81 @@ exports.log = log;
 exports.Models = Models;
 exports.IDs = IDs;
 
-before(function before(next) {
-	this.timeout(60 * 1000);
+// NOTE: SKIPPED UNTIL EXTERNAL SERVICES ARE RESOLVED
```
Comment (todanator) [inline test/common.js]: This is until we can sort out how to do integration tests better right? If so I would suggest putting a link to our JIRA ticket

File: test/inner_join.js [hunk 3]
```diff
@@ -12,65 +12,65 @@ describe('Inner Join', function () {
...
-		});
-		ChildModel = Arrow.Model.extend('childModel', {
-			fields: {rid: {type: Number}, languages: {type: Object}},
-			connector: 'memory'
-		});
-		ChildModel2 = Arrow.Model.extend('childModel2', {
-			fields: {rid: {type: Number}, nationality: {type: String}},
-			connector: 'memory'
-		});
+	// before(function (done) {
```
Comment (todanator) [inline test/inner_join.js]: Same, you can probably do a 'skip' on the 'describe'

File: test/joins_one_to_many_test.js [hunk 4]
```diff
@@ -0,0 +1,228 @@
+var should = require('should');
+var async = require('async');
+var utils = require('./utils')();
+
+describe('Composite - One to Many', function () {
```
Comment (todanator) [inline test/joins_one_to_many_test.js]: It seems like the left join tests should go in left_join.js and the inner join tests should go in inner_join.js

File: test/inner_join.js [hunk 5]
```diff
@@ -12,65 +12,65 @@ describe('Inner Join', function () {
...
+	// 				join_properties: {
+	// 					rid: 'rid'
+	// 				}
+	// 			}]
+	// 		}
+	// 	});
+	// });
 
-	it('API-285: should support inner join', function (next) {
+	it.skip('API-285: should support inner join', function (next) {
```
Comment (todanator) [inline test/inner_join.js]: Couldn't these tests be updated to work? They depend on memory connector just like Author/Post.

File: test/joins_one_to_many_test.js [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+			should(result[1].posts.length).equal(1);
+			should(result[2].posts).be.not.ok();
+			return next();
+		});
+	});
+
+	/**
+	 * Multiple has no effect here and we get the same as previous
+	 */
+	it('[RDPP-4451] Left Join Array - No Alias - With Multiple', function (next) {
```
Comment (todanator) [inline test/joins_one_to_many_test.js]: These don't appear to be generated testids.

File: test/left_join.js [hunk 7]
```diff
@@ -8,7 +8,7 @@ describe('Left Join', function () {
 	var Models = common.Models,
 		IDs = common.IDs;
 
-	it('should warn when joined on a bad model', function (next) {
+	it.skip('should warn when joined on a bad model', function (next) {
```
Comment (todanator) [inline test/left_join.js]: Here too, seems like these should be updated to use the arrow server in your utils.js.

**Review by todanator — CHANGES_REQUESTED** [review]

File: test/joins_one_to_many_test.js (hunk 4 above)
Comment (pltod2) [reply test/joins_one_to_many_test.js]: You are right but I will approach it differently. I will just delete those one to remove confusion. All the cases are covered by the new classification one to many and one to one. I believe such classification is much more useful than inner - left join.

File: test/joins_one_to_many_test.js (hunk 6 above)
Comment (pltod2) [reply test/joins_one_to_many_test.js]: Yes. They are manual we do not have script for this project that supports automatic generation

File: test/batched.js (hunk 1 above)
Comment (pltod2) [reply test/batched.js]: No skips anymore. Old code is in separate suite that is not considered.

File: test/common.js (hunk 2 above)
Comment (pltod2) [reply test/common.js]: Comment added in dedicated readme

File: test/inner_join.js (hunk 5 above)
Comment (pltod2) [reply test/inner_join.js]: These are not going to be needed anymore unless they are integration tests

File: test/left_join.js (hunk 7 above)
Comment (pltod2) [reply test/left_join.js]: These are not going to be needed anymore unless they are integration tests

**Pull request merged**